lang/py3/setup.py

```diff
@@ -93,10 +94,9 @@ def SetupSources():
   )
 
   # Make sure the avro shell script is executable:
-  os.chmod(
-      path=os.path.join(py3_dir, 'scripts', 'avro'),
-      mode=0o777,
-  )
+  script = os.path.join(py3_dir, 'scripts', 'avro')
+  os.chmod(script,
+           stat.S_IMODE(os.stat(script).st_mode) | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
```

Comment:
I'm not convinced this chmod is necessary at all, but this does what the comment says.

Reply:
Why is this needed?

Reply:
Please do!